packages/extension-ui/src/messaging.ts

```diff
@@ -152,8 +153,21 @@ export async function getMetadata (genesisHash?: string | null, isPartial = fals
   }
 
   const def = await request;
-
   if (def) {
+    const specVersionInState = def.specVersion;
+    // when spec version is not the latest, fetch latest and updated state with latest metadata
+    if (specVersion && !specVersion.eqn(specVersionInState)) {
+      console.log('Spec versions are different - update meta')
+      const metaDataInfo = getLatestMetaFromServer(genesisHash);
+      const newTypes = getLatestTypesFromServer();
+      if (metaDataInfo) {
+        def.specVersion = metaDataInfo.specVersion;
+        def.metaCalls = metaDataInfo.metaCalls;
+        const oldTypes = def.types;
+        def.types = {...oldTypes, ...newTypes};
+        await sendMessage('pri(metadata.set)', def);
+      }
+    }
```

Comment:
can this fail? if so what should user see?

Reply:
no.. won't fail... If the http request fails for any reason.. the catch will send null as the output from getLatestMetaFromServer and it won't do any further processing and will work as a normal flow